AdventOfCode2016/Day1.fs

```diff
@@ -0,0 +1,84 @@
+﻿module Day1
+
+open System
+open System.IO
+
+type Turn = Left | Right
+
+type Direction = North | East | West | South
+
+type Instruction = {
+    Direction: Turn;
+    Distance: int;
+}
+
+type Coordinate = {
+    X : int;
+    Y : int;
+}
+
+type State = {
+    Position: Coordinate;
+    Facing: Direction;
+}
+
+let parseDirection (c:char) =
+    match c with
+    | 'L' -> Left
+    | 'R' -> Right
+    | _ -> raise <| new System.FormatException(sprintf "Could not parse %c" c)
+
+let parseInstruction (s:string) =
+    {
+        Direction = parseDirection s.[0]
+        Distance = Int32.Parse(s.Substring(1))
+    }
+
+let left facing =
+    match facing with
+    | North -> West
+    | West -> South
+    | South -> East
+    | East -> North
+
+let right facing =
+    match facing with
+    | North -> East
+    | East -> South
+    | South -> West
+    | West -> North
+
+let move position direction distance =

```

Comment:
The pattern I've usually seen is to let inference exist in all cases possible. If you're creating a library for external publication, add `.fsi` files with type signatures to clarify how it works.

I'm not 100% on-board with that approach, but it's the most common.